src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -20,9 +20,9 @@ namespace Microsoft.CodeAnalysis.CodeActions
     public abstract class CodeAction
     {
         /// <summary>
-        /// A short title describing the action that may appear in a menu.
+        /// A short localizable title describing the action that may appear in a menu.
         /// </summary>
-        public abstract string Title { get; }
+        public abstract LocalizableString Title { get; }
```

Comment:
One other alternative API design is to have a protected constructor for CodeAction that takes the LocalizableString title as parameter, and stores `title.MakeExceptionSafe()` result onto a readonly field on CodeAction. Title can then become a non-abstract property and issue 1. would be avoided.

You can also fix issue 2. above by registering a handler for CodeAction.Title.OnException event in CodeFixService. CodeFixService can MEF import AbstractHostDiagnosticUpdateSource, and invoke it's report diagnostic method with exception diagnostic. Probably issue 2. needs a separate work item and can be done later.